deepreg/vis.py

```diff
@@ -0,0 +1,370 @@
+        imgs.append(img)
+
+    for img, col_num in zip(imgs, range(num_imgs)):
+        for index, row_num in zip(slice_inds, range(num_inds)):
+            plt.subplot(num_inds, num_imgs, subplot_mat[row_num, col_num])
+            plt.imshow(img[:, :, index])
+            plt.axis("off")
+            if row_num - 0 < 1e-3:
+                plt.title(col_titles[col_num])
+
+    if fname is None:
+        fname = "visualisation.png"
+    save_fig_to = os.path.join(save_path, fname)
+    plt.savefig(save_fig_to)
+    print("Plot saved to:", save_fig_to)
+
+
+def gif_warp(
+    img_paths, ddf_path, slice_inds=None, num_interval=100, interval=50, save_path=""
+):
+    """
+    Apply ddf to image slice/s to generate gif
+
+    :param img_paths : list or comma separated string of image paths
+    :param ddf_path : path to ddf to use for warping
+    :param slice_inds : list of slice indices to use for each image
+    :param num_interval : number of intervals in which to apply ddf
+    :param interval : time in miliseconds between frames of gif
+    :param save_path : path to directory where visualisation/s is/are to be saved
+    """
+    if type(img_paths) is str:
+        img_paths = string_to_list(img_paths)
+
+    image = load_nifti_file(img_paths[0])
+    img_shape = np.shape(image)
+
+    if slice_inds is None:
+        slice_inds = [round(np.random.rand() * (img_shape[-1]) - 1)]
+
+    for img_path in img_paths:
+        for slice_ind in slice_inds:
+
+            fig = plt.figure()
+            ax = plt.Axes(fig, [0.0, 0.0, 1.0, 1.0])
+            ax.set_axis_off()
+            fig.add_axes(ax)
+
+            ddf_scalers = np.linspace(0, 1, num=num_interval)
+
+            frames = []
+            for ddf_scaler in ddf_scalers:
+                image = load_nifti_file(img_path)
+                ddf = load_nifti_file(ddf_path)
+                image = np.expand_dims(image, axis=0)
+                ddf = np.expand_dims(ddf, axis=0) * ddf_scaler
+
+                warped_image = warp_image_ddf(image=image, ddf=ddf, grid_ref=None)
+                warped_image = np.squeeze(warped_image.numpy())
+
+                frame = plt.imshow(
+                    warped_image[:, :, slice_ind], aspect="auto", animated=True
+                )
+
+                frames.append([frame])
+
+            anim = animation.ArtistAnimation(fig, frames, interval=interval)
+            path_to_anim_save = os.path.join(
+                save_path,
+                os.path.split(img_path)[-1].split(".")[0]
+                + "_slice_"
+                + str(slice_ind)
+                + ".gif",
+            )
+
+            anim.save(path_to_anim_save)
+            print("Animation saved to:", path_to_anim_save)
+
+
+def gif_tile_slices(img_paths, save_path=None, size=(2, 2), fname=None, interval=50):
+    """
+    Creates tiled gif over slices of multiple images
+
+    :param img_paths : list or comma separated string of image paths
```

Comment:
I'm just wondering if `:param xxx :` works, there's a space before `:`

Reply:
Sorry, that's a typo. I will correct it now